### PR TITLE
Change sqlite to mongodb

### LINKED ISSUE #19: [Matching service] Change sqlite to mongodb
Discussion had to change sqlite to mongodb, reasons:

- sqlite is mainly local, mongodb can be local and cloud (cloud db to go for)

### PR DESCRIPTION
**Pull request by tanwayne890**
Closes #19 

As the title suggests, this PR adds:
- instruction to readme
- `model` directory to contain schema, orm functions and mongoose model function (file structures are referenced from user-service)
- a match-controller into `controllers`

Video showing 2 users login to `user-service` and match up (sent to tele, too large to put here)

### CODE REVIEW

**Review by oliviajohansen — APPROVED** [review]
just some suggestions

File: matching-service/src/model/match-orm.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        return true;
+
+    } catch (err) {
+        console.log('ERROR: Could not create new user difficulty');
+        return { err };
+    }
+}
+
+export async function ormDeleteUserDifficulty(userId) {
+    if (!await existsUserDifficulty(userId)) {
```
[the remark this answers is not in the document]
Comment (tanwayne890) [reply matching-service/src/model/match-orm.js]: https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g23/pull/18/commits/ffc15fdc149caee602b0c8f9b85c8820e1a46f69

File: matching-service/src/model/repository.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+let mongoDB = process.env.ENV == "PROD" ? process.env.DB_CLOUD_URI : process.env.DB_LOCAL_URI;
+
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+
+
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+export async function createUserDifficulty(params) {
```
[the remark this answers is not in the document]
Comment (tanwayne890) [reply matching-service/src/model/repository.js]: https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g23/pull/18/commits/73f0790c34d194c612c7dc2caad8d32675e93707

File: matching-service/src/model/match-orm.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { createUserDifficulty, deleteUserDifficulty, existsUserDifficulty } from './repository.js';
+import "dotenv/config";
+
+//need to separate orm functions from repository to decouple business logic from persistence
+export async function ormCreateUserDifficulty(userId, difficulty) {
```
[the remark this answers is not in the document]
Comment (tanwayne890) [reply matching-service/src/model/match-orm.js]: https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g23/pull/18/commits/81b3facc753ceb375c3dc7482540b102fbcdef96

[the remark this answers is not in the document]
Comment (oliviajohansen) [reply matching-service/src/model/match-orm.js]: thanks

File: matching-service/src/model/match-orm.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (oliviajohansen) [reply matching-service/src/model/match-orm.js]: thanks

File: matching-service/src/model/repository.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliviajohansen) [reply matching-service/src/model/repository.js]: thanks

**Pull request merged**